### PR TITLE
Add client-server architecture drafts

### LINKED ISSUE #67: Create our first draft of Client-Server Architecture Design
Create our first draft of client-server architecture design as per the [C4 Model](https://github.com/holtzmak/Humane-Transport/wiki/C4-Model-for-Client-Server-Architecture). 

_We say first draft here as we'll expect to expand on the design, or expect the design to expand, in the future with the addition of other systems the application will communicate with (like more than one database)._

This task is to start designing with the one database and application in mind. There are 4 Levels of abstractions (drawings) in the C4 Model:
1. System Context Diagram. Shows how the application interacts with systems around it (the database!)
1. Container Diagram. Shows the major groups of classes within the application (think GUI classes, Form-handling classes, Database interaction classes, etc.)
1. Component Diagram. Shows the breakdown of the major groups of classes, and their classes. (What classes do we envision create our GUI-related code?)

**We won't do Code Diagrams at this time as code-based drawings change too frequently to bother with during initial design and before programming**. *However, exceptions do exists, we'll make this diagram only if some foreseen class is particularly interesting and the diagram would benefit us right now*
4. Code Diagrams. UML Diagrams, Entity Diagram, etc.

Brainstorm the responsibilities of the application as per the client-server knowledge you may have from job or university experiences. We'll generate classes with responsibilities and other good code practices in mind (like the Single Responsibility Rule (SRP) and Don't Repeat Yourself (DRY)). 

This task is done when (depending on how the work feels, each of these could be split to it's own task):
- [x] The first draft of the System Context Diagram is done (with the one database in mind)
- [x] The first draft of the Container Diagram is done (with responsibilities as we see/understand now)
- [x] The first draft of the Component Diagram is done  (with responsibilities as we see/understand now)

Final note: First draft is an important word in this task, when working with Flutter we'll probably find that the classes we thought we'd use are totally different in Flutter/dart (but the responsibilities stay the same). Thus, when we have made progress in Flutter, we can go back and create another draft of the architecture design with a better idea of it. 

*I do foresee that going back into the architecture drawings could slow down development, hence why I've mentioned before that if we have the general idea, we can forge forward without redrawing design (the code is now our design); We'll revisit the design drawings when we have the time.*

### PR DESCRIPTION
**Pull request by holtzmak**
These drawings are per the C4Model investigated in #27 . 

Please review in order of the Context Diagram, then Container Diagram - Humane Transport, Container Diagram - Mobile App, then the Component diagrams.

These are not the final drawings of the architecture, I will move these to UML Diagrams using Draw.io when they are discussed and approved in this PR.

Closes #67 .

### CODE REVIEW

**Comment by holtzmak** [comment]
@MummenRider , @mansiPaTeLL , @SanaKhan1997 updated the drawings to UML, PDF format for easier reading. Ready for review still.

File: documentation/client-server architecture/Client Component Diagram.drawio [hunk 1]
```diff
@@ -0,0 +1 @@
...wfEzjquZvCulpRpTiKljAbqmzkSyZtE+7nD5RvtrQPRXBEp7gsPWimBT+MqcEw8Okkx+n6bDPiCmzbLP7XF1AG87B2DTdojSOhDLarcaxMcaeEO/7TM2u3dpmteMCjcwmde36QxNTdzJjBHQPOM215yvDpFqr1jQNPT11aIaD4jkqeaKEpGJolXzulELZghuZUJXB+ig5wHZhqHkrSmIVsyp5dWEoGzeZG7RvihXNrUH8xqRQnShqsU9zu7QaL2bnU80jq+IIyO3Ut0OdVXGsAMZOuTovnTMfvL4RXO0gCB32a6dCzRi0f94Sg/bOfj4Gfdt1/CgElqLVPcQkhBBFzcJfqCdivzLn7dvqXpMuJqVYDwOK46K/6MRFmi369YL7S313UxFNUXtAAvS9cJJE1U7t5fraexyN7vrX958ffv/9zz+/PrJgybrO/qyvGxxrod1KtNXI5liHsmOZT6hHqh0I1XKFHzbZ9ugs5lVkO2C0VVZclS2AI+kOgXSr9H/Y4OsfCPj6Bws+s9125NxhcC5T92FjbY9Ogl+FtcE7xdrWDNvuI4kv1KNMv/5Qfr9Mvf2wmmVbeOSW1ozHSB18iVDgKdld/WOHulnfPjzcb2Gkw7VGSkN/hN/gQGrChfe9ZpW1d/9XvQbb/NYA6r1hfPP3E5RPVOPZi7mdVefr+ZF66/k6GHL8XHoIk/mpPCaKx1Sqei5fOzTfeFC/+rWQ6kH9xu2tpqtZp33XPqt4W6/ha+veAN76fN904R7foO6UXvqouXq/5sFaDuapwokbFbmbKtKCWlNRVlBMpwmtlFHEyGX3Soi80v13xKB3DpHzI0TeFUTKkUcbRHr98zeESH9Q9f3e4IUQGZxtqOgIkT2CiG11Dosi/YFd5oh1CjH6Bpao1H32oebbAaa8Vd0apTjD82OUcgTM+wLM+s8J/3mAee9hSnnftzVMsV274n+DI0WOFHlripRX6EeK7IQiq3dCqhTZ+AljkyJd69SxMnBkwcjbRiMvXe40OHJc7uwzR9a/7nHkyK440tsZR2prml8ajdj1Q5b6r7XZliL9+pfQx2hknymy/t2JI0V2RZH+rigCYq3uvXaHvzQaeT8cWbYO4UiVd0CV9a8uHKmyK6q07JS8kCqDs2pw0nXOfyVW3PeDlR2CBJLFr/fUTxe/JLV3/T8=</diagram></mxfile>
```
Comment (holtzmak) [inline documentation/client-server architecture/Client Component Diagram.drawio:1]:
```suggestion
<mxfile host="Electron" modified="2020-12-12T22:02:24.926Z" agent="5.0 (Macintosh; Intel Mac OS X 10_16_0) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/13.9.9 Chrome/85.0.4183.121 Electron/10.1.5 Safari/537.36" etag="EkC9kN79lQDUtd3sMLYt" version="13.9.9" type="device"><diagram id="esmy7CawzLTuU-0gui4n" name="Page-1">7ZxZc9s4EoB/jaqcB7lISpTlR/mKXZvMeiJ7kp2XLYiEJExAgiFBS55fv90AeFNHbG1sjRSnEgIEcfTxoQGQ7vQug+XHmETzz8KnvONY/rLTu+o4ju2encF/mPOsc84HPZ0xi5lvChUZY/Y3NZmWyU2ZT5NKQSkElyyqZnoiDKknK3kkjsWiWmwqeLXViMxoI2PsEd7M/cp8Ode5Q9cq8m8pm82zlm3L3AlIVthkJHPii0Upq3fd6V3GQkh9FSwvKUfhZXLRz92suJt3LKahbHlATP5CeTgWJxPQiSrQcQYcyl5MRYh3Evlshjn4kYrsRjdRShhBAduKlsVNuJqZ/1UtkyzjkjOqKrwUQSRCda2LQG8mjcfiVzTtXhRtuFelZvSQspaUZDjx6Bz0TeOqDYQkMModi6lcgAAhc/ycSBrAxW94V5XLJH3zxMf81ubet6+3Z3R5K+zfvnXdXIO5ZooxObFIQ5/i8xb0ZzFnko4j6A5kLMBPIG8uA1SKDZeJjMV3eim4iNXTvaGDPygTxnmWH8KgjZhKRS31B/IJZ7MQ8jidogiU0u9FwiQTmO2BwCg8c/FEY8nAvD/VCgTM97H3eYGRqXAipBQB3PBJMldDKvqceQTmRDRmAYVGcKAsnJmxJ3nKKVIPIjK3cTjG6W3LNHJPJFSDbQ9BqsNCoSVTN6rEztJlKcto5CMV0Jf4GYqYu5kXGgzZZya9KDl1xpx5yaFzEBFjRLO86sLX4MJYQZY03rfSE0ueYl0xMotJoOiEviHnaJBtTmWJKfzzWUwYxyKjKOKgKqVBLBdKwkLQcsnRqq4H4g9bnc/TFoWOF88mJzBk+AvjsEpXH/AS67KUqy6MkPCZPpZbhYmtYWO3efxXqjwRhs+mDMwPKiqc1stEk0BigtqOaQI5CQMJMflcEylDH5hqe4TyVC4oRXl8fLxDBYdY+5jGTyjCQuzJ6UrM5NlasC/h3LKbBETh4yWyXanQnbTyiSRS2ZuvZa9KXlGPBhMlItvRpoE/K4WxJYoLJygADBO/F7PIEAoLlTI2I7pnr2U0gENWQQzGw/4mE1UABRUJhqYFpd0LnG2AsqkUicFVE7p1dErEXJV53X4Nelm6xPS+hT9ZOdPhlun+tQx0B00Gum4LA93/CwIzi/HZ00vN2u5vFZ+o0BQVA488Jsp07wwK6LpIZYVjQ3ZLn18xjOHGUEcVu+EpTow5sEb3d9UY6A07u3LKeW23stkjWxhkLdvNTow4x3DfsdIEOYMLhQz6RBnfgoGZZ5MBqUyeGv4B+U7VJPIjpYmaVPT04cUUCuKsgbh7YnShp5DRw5ckf7jIJp4H8Z9UjUMFgWlk0yRSGf2W2CwHBwU4JfXmIQBlZjydns5OcW6LYt3BC0rCn0Xsp/9ak8fgzB//69/8x8fxsjvp/9Ht7S4KLse6GAO7EzLNINgW8Dai5sH5uXNx1YLxHZCz13Or4WOTnPawhZy2s+fobIZDewTMKxLLIyFLnbgFUnGa5DNICXSaYUArCK19zAjpog6wElBDNZlW4m9dAwRV2d1ELanx1gk7pQighDyp1jWBgZ7YZCyCoiuasGmiOVU0ddOs98M/kqg73FfYK6L2nV+H1MZeWqsmbNPBkvipP6NZQC5iORczASZ5XeReVBVUlPkkVPiPavmLSvlswn9cT1SVRpdMfsPHT12T+k/pztXS1KwSz5vUkYg09uimXSywbhLPqNxcDse/Vrkx5cCKp+re6U8oakvNDI21EZ7S6ow4t/No0PfV3lYW9U9TruL+toCseCrLjN7X/oM1FkgiKxSSLtCqsGpm6Ip9wYlnzQQT1fMEr+dw1jq3cRbS7ry0U6D8z23OLdfEw+Ba7VCZHSzgTCw418G4lliEU3Kl1Oc/LlW+3vNT+NcRuRJ1QBdzoqMfMFG1o6L25lW5xZyqSnVFGH+jVIoJRsvVK4B+md/yWvrW2mulFNQq0AMNu1VDumZdVtcS4EFIUTWFlQctDKE0756wEGYp5qfKTpmZ2KyS1VwRSSYkwYnzZjyH+rSZSFyQJLVuWCczIFCUZCJm9YkStdzUu855WCmZkHo0SYhycVKED5qpJQE+R7jK5so4cx2yIFK7ldrxcuXj7otZTvlUGAmmPjOXSgAsIDMVMHRR2w1P1PIJqU5gbKJGb8yMeSkYTclPxLSwOj0AcEh8Ciipq1Y6UGHJZUXGmBplizpMXAeE8bLWhdFvQqFpIrGJk2kq05iiNqewfEzjquZvCulpRpTiKljAbqmzkSyZtE+7nD5RvtrQPRXBEp7gsPWimBT+MqcEw8Okkx+n6bDPiCmzbLP7XF1AG87B2DTdojSOhDLarcaxMcaeEO/7TM2u3dpmteMCjcwmde36QxNTdzJjBHQPOM215yvDpFqr1jQNPT11aIaD4jkqeaKEpGJolXzulELZghuZUJXB+ig5wHZhqHkrSmIVsyp5dWEoGzeZG7RvihXNrUH8xqRQnShqsU9zu7QaL2bnU80jq+IIyO3Ut0OdVXGsAMZOuTovnTMfvL4RXO0gCB32a6dCzRi0f94Sg/bOfj4Gfdt1/CgElqLVPcQkhBBFzcJfqCdivzLn7dvqXpMuJqVYDwOK46K/6MRFmi369YL7S313UxFNUXtAAvS9cJJE1U7t5fraexyN7vrX958ffv/9zz+/PrJgybrO/qyvGxxrod1KtNXI5liHsmOZT6hHqh0I1XKFHzbZ9ugs5lVkO2C0VVZclS2AI+kOgXSr9H/Y4OsfCPj6Bws+s9125NxhcC5T92FjbY9Ogl+FtcE7xdrWDNvuI4kv1KNMv/5Qfr9Mvf2wmmVbeOSW1ozHSB18iVDgKdld/WOHulnfPjzcb2Gkw7VGSkN/hN/gQGrChfe9ZpW1d/9XvQbb/NYA6r1hfPP3E5RPVOPZi7mdVefr+ZF66/k6GHL8XHoIk/mpPCaKx1Sqei5fOzTfeFC/+rWQ6kH9xu2tpqtZp33XPqt4W6/ha+veAN76fN904R7foO6UXvqouXq/5sFaDuapwokbFbmbKtKCWlNRVlBMpwmtlFHEyGX3Soi80v13xKB3DpHzI0TeFUTKkUcbRHr98zeESH9Q9f3e4IUQGZxtqOgIkT2CiG11Dosi/YFd5oh1CjH6Bpao1H32oebbAaa8Vd0apTjD82OUcgTM+wLM+s8J/3mAee9hSnnftzVMsV274n+DI0WOFHlripRX6EeK7IQiq3dCqhTZ+AljkyJd69SxMnBkwcjbRiMvXe40OHJc7uwzR9a/7nHkyK440tsZR2prml8ajdj1Q5b6r7XZliL9+pfQx2hknymy/t2JI0V2RZH+rigCYq3uvXaHvzQaeT8cWbYO4UiVd0CV9a8uHKmyK6q07JS8kCqDs2pw0nXOfyVW3PeDlR2CBJLFr/fUTxe/JLV3/T8=</diagram></mxfile>
```

**Review by MummenRider — COMMENTED** [review]
I checked all diagrams and they're all perfect.

**Review by MummenRider — APPROVED** [review]
These files are ready to be merge.

**Pull request merged**